### PR TITLE
Add com.obsproject.Studio.Plugin.DistroAV

### PR DESCRIPTION
**Pull request by tytan652**
### Please confirm your submission meets all the criteria

- [x] Please describe the ~~application~~ extension briefly.
  
  Network A/V in OBS Studio with NewTek's NDI technology in a Flatpak.
  
  Resubmission of [com.obsproject.Studio.Plugin.NDI](https://github.com/flathub/com.obsproject.Studio.Plugin.NDI) (#4701) as DistroAV, the plugin has rebranded to remove OBS and NDI from their name.

- [x] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
  - This is an OBS Studio plugin
- [x] I have read the and followed all the [Submission and licence requirements][reqs].
- [x] I have [built][build] and tested the submission locally.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission.
   - Request: https://github.com/obs-ndi/obs-ndi/issues/724#issuecomment-1808136817
  - Answer: https://github.com/obs-ndi/obs-ndi/issues/724#issuecomment-1812959827
  - The upstream contributor @paulpv wishes to have access to the repo, and also asked @Palakis to be added.
  - **Please do not forget to re-add them to new repo** (#4707)

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

**Comment by tytan652** [comment]
I forgot to update the plugin to version 6 (which has the rebrand).

File: .github/workflows/check-ndi-sdk.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      - uses: docker://ghcr.io/flathub/flatpak-external-data-checker:latest
+        env:
+          GIT_AUTHOR_NAME: Flatpak External Data Checker
+          GIT_COMMITTER_NAME: Flatpak External Data Checker
+          GIT_AUTHOR_EMAIL: 41898282+github-actions[bot]@users.noreply.github.com
+          GIT_COMMITTER_EMAIL: 41898282+github-actions[bot]@users.noreply.github.com
+          EMAIL: 41898282+github-actions[bot]@users.noreply.github.com
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          args: --filter-type  ${{ matrix.filter-type }} --update --never-fork com.obsproject.Studio.Plugin.DistroAV.json
```
Comment (bbhtt) [inline .github/workflows/check-ndi-sdk.yaml:29]: matrix.filter-type doesn't seem to exist

File: .github/workflows/check-ndi-sdk.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  - cron: "* * * * 1"
+  workflow_dispatch: 
+
+jobs:
+  flatpak-external-data-checker:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        branch: [ master ]
```
Comment (bbhtt) [inline .github/workflows/check-ndi-sdk.yaml:13]: if there is only one branch why is this still needed? You can use the global x-checker?

**Comment by bbhtt** [comment]
bot, build com.obsproject.Studio.Plugin.DistroAV

File: .github/workflows/check-ndi-sdk.yaml (hunk 2 above)
Comment (tytan652) [reply .github/workflows/check-ndi-sdk.yaml]: It was from the example itself.

File: .github/workflows/check-ndi-sdk.yaml (hunk 1 above)
Comment (tytan652) [reply .github/workflows/check-ndi-sdk.yaml:29]: Need to set it, I forgot to do it on the NDI repo… which explain things.

**Comment by paulpv** [comment]
I'd prefer the package name not have `obsproject` in it.
The entire point of the rebrand was for us to not be confused as part of the OBS project/team.
I prefer something more like:
`org.DistroAV.DistroAVOBS`.
The rationale for the OBS suffix is that we're might one day release versions for other casting software.
All lower case is fine with me. 
I am guessing we cannot have `-` dashes.

**Comment by tytan652** [comment]
@paulpv, this is not how Flatpak plugins works. Their id must be a sub domain or the plugin endpoint (here `com.obsproject.Studio.Plugin`).

@bbhtt sorry I copied the wrong action file over the PR, now it should be good.

**Comment by bbhtt** [comment]
You can set extension points that don't match the prefix in OBS manifest but you would need to do it for every new plug in which is problematic and requires changing the OBS manifest and pushing new builds every time someone wants a plug in submitted.

There is no good reason to go in that mess.

**Comment by paulpv** [comment]
@tytan652 @bbhtt Understood. Name as is is fine then.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# [DistroAV](https://github.com/DistroAV/DistroAV) Flatpak
+
+Network A/V in OBS Studio with NewTek's NDI technology in a Flatpak.
+
+## Notes
+Make sure that `--system-talk-name=org.freedesktop.Avahi` is set in OBS Studio permissions. Otherwise obs-ndi will not be able to see other devices or send to the network, since NDI on Linux rely on the Avahi daemon for network discovery (find other NDI device on the network).
```
Comment (paulpv) [inline README.md:6]: obs-ndi -> DistroAV

File: com.obsproject.Studio.Plugin.DistroAV.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  <id>com.obsproject.Studio.Plugin.DistroAV</id>
+  <provides>
+    <id>com.obsproject.Studio.Plugin.NDI</id>
+  </provides>
+  <replaces>
+    <id>com.obsproject.Studio.Plugin.NDI</id>
+  </replaces>
+  <extends>com.obsproject.Studio</extends>
+  <name>DistroAV</name>
+  <developer>Paul Peavyhouse</developer>
```
Comment (paulpv) [inline com.obsproject.Studio.Plugin.DistroAV.metainfo.xml:12]: distroav@distroav.org

File: README.md (hunk 3 above)
Comment (tytan652) [reply README.md]: Fixed

File: com.obsproject.Studio.Plugin.DistroAV.metainfo.xml (hunk 4 above)
Comment (tytan652) [reply com.obsproject.Studio.Plugin.DistroAV.metainfo.xml]: A mail can't be used here, it needs to be legible.

But I replaced the author by DistroAV, newer metainfo rules that I forgot to apply ask for an id (usually a reverse DNS).
https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-developer

**Comment by bbhtt** [comment]
bot, build com.obsproject.Studio.Plugin.DistroAV

**Pull request closed without merging**